core/services/relay/evm/capabilities/workflows/syncer/workflow_syncer_test.go

```diff
@@ -0,0 +1,234 @@
+package workflow_registry_syncer_test
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services/servicetest"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/workflow/generated/workflow_registry_wrapper"
+	coretestutils "github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils/pgtest"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/capabilities/testutils"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/services/workflows/syncer"
+	"github.com/smartcontractkit/chainlink/v2/core/utils/crypto"
+	"github.com/smartcontractkit/chainlink/v2/core/utils/signalers"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_SecretsWorker(t *testing.T) {
+	var (
+		ctx       = coretestutils.Context(t)
+		lggr      = logger.TestLogger(t)
+		backendTH = testutils.NewEVMBackendTH(t)
+		db        = pgtest.NewSqlxDB(t)
+		orm       = syncer.NewWorkflowRegistryDS(db, lggr)
+
+		giveTicker     = signalers.MakeTicker(ctx.Done(), 500*time.Millisecond)
```

Comment:
What does this helper do? Would something go wrong if the ticker was used directly instead?
```suggestion
		ticker     := time.NewTicker(500*time.Millisecond)
		t.Cleanup(ticker.Stop)
```

Reply:
the ticker on the worker isn't actually a time ticker, it's a `chan struct{}`.  This helper turns a time interval into a `chan struct{}`.  in smaller unit tests we can just send the structs manually to trigger side effects, but with the log poller as a dependency the worker has to repeatedly check what's been polled.

Reply:
But why not just use the standard `chan time.Time` ticker channel? Why is it important or beneficial to force a conversion to another type?